### ISSUE TITLE
(PostgreSQL) deadlock detected - Process 10326 waits for ShareLock

### ISSUE THREAD

**Issue by grooviegermanikus** [blockworks-foundation/BankingStageErrorsTrackingSidecar#58]
### Problem

Database log shows this -- while the sidecar gets stuck:
```
2024-02-08 14:15:32.131 CET [10326] bankingstage_sidecar@banking_stage ERROR:  deadlock detected
2024-02-08 14:15:32.131 CET [10326] bankingstage_sidecar@banking_stage DETAIL:  Process 10326 waits for Share
Lock on transaction 2245819; blocked by process 10328.
        Process 10328 waits for ShareLock on transaction 2245818; blocked by process 10326.
        Process 10326:
                INSERT INTO banking_stage_results_2.accounts(account_key) SELECT key from temp_table_2_22322
                ON CONFLICT DO NOTHING

        Process 10328:
                INSERT INTO banking_stage_results_2.accounts(account_key) SELECT key from temp_table_4_22308
                ON CONFLICT DO NOTHING
```

same happens for most other tables too !

**Comment by grooviegermanikus** [comment]
reproduce locally:


```SQL

```



INSERT INTO sample_lookup(acc_id, account_key)
SELECT t.id::bigint as acc_id, 'foo'||id FROM generate_series (1,5000000) as t(id)
order by id asc
ON CONFLICT DO NOTHING

**Comment by grooviegermanikus** [comment]
explained [here](https://dba.stackexchange.com/questions/194756/deadlock-with-multi-row-inserts-despite-on-conflict-do-nothing) and [here](and https://rcoh.me/posts/postgres-unique-constraints-deadlock/)

**TL;DR:** use "Lock Ordering" [(c2)](https://wiki.c2.com/?OrderedLocks)

**Comment by grooviegermanikus** [comment]
```
        Process 13292 waits for ShareLock on transaction 2490306; blocked by process 13290.
        Process 13290:
                    INSERT INTO banking_stage_results_2.accounts_map_transaction(acc_id, transaction_id, is_w
ritable, is_signer, is_atl)
                        SELECT
                            ( select acc_id from banking_stage_results_2.accounts where account_key = new_amt
_data.account_key ),
                            ( select transaction_id from banking_stage_results_2.transactions where signature
 = new_amt_data.signature ),
                            new_amt_data.is_writable,
                            new_amt_data.is_signer,
                            new_amt_data.is_atl
                        FROM temp_table_2_3087 AS new_amt_data
                        ON CONFLICT DO NOTHING
```
